telegram-interaction.js

```diff
@@ -0,0 +1,349 @@
+    Shelly.call(
+      "KVS.Get", 
+      { key: CONFIG.identName },
+      (function (data, error) {
+        let value = 0;
+        if(error !== 0) {
+          console.log("Cannot read the value for the provided key, reason, using default value.");
+        }
+        else {
+          value = data.value;
+        }
+        this.messageOffset = value;
+        Shelly.emitEvent(CONFIG.identName);
+      }.bind(this))
+    );
+  },
+  
+  /**
+   * Called when the event specified in the CONFIG is emitted
+   */
+  onEvent: function () {
+    if(CONFIG.debug) {
+      console.log("Poll started");
+    }
+    Shelly.call(
+      "HTTP.REQUEST",
+      { 
+        method: "POST",
+        url: "https://api.telegram.org/bot" + CONFIG.botKey + "/getUpdates", 
+        timeout: CONFIG.timeout,
+        body: {
+          offset: this.messageOffset + 1,
+          limit: 1
+        }
+      },
+      this.onFinishPoll.bind(this),
+      this
+    );
+  },
+
+  /**
+   * Callback function after finishing polling updates from the Telegram API. See https://shelly-api-docs.shelly.cloud/gen2/Scripts/ShellyScriptLanguageFeatures#shellycall
+   * @param {Object|null|undefined} data the received data from the request  
+   * @param {Number} error the id of the error 
+   * @param {*} errorMessage the error message if has
+   * @returns 
+   */
+  onFinishPoll: function (data, error, errorMessage) {
+    if(error !== 0) {
+      console.log("Poll finishes with error ->", errorMessage);
+      console.log("Aborting...");
+      return;
+    }
+
+    let response = JSON.parse(data.body);
+
+    if(!response.ok) {
+      console.log("Something is wrong with the config. Error:", response.description);
+      return;
+    }
+
+    if(response.result.length === 0 && CONFIG.debug) {
+      console.log("No new messages");
+    }
+
+    let lastUpdateId = -1;
+    for (let res of response.result) {
+      if(CONFIG.debug) {
+        console.log("New message", res.message.text);
+      }
+      this.handleMessage(res.message);
+      lastUpdateId = res.update_id;
+    }
+
+    if (lastUpdateId > this.messageOffset) {
+      this.messageOffset = lastUpdateId;
+      Shelly.call("KVS.Set", { key: CONFIG.identName, value: this.messageOffset } );
+    }
+
+    Timer.set(CONFIG.timer, false, function() {
+      Shelly.emitEvent(CONFIG.identName);
+    });
+  },
+
+  /**
+   * Processes received messages
+   * @param {Object} message received message object from the API
+   */
+  handleMessage: function (message) {
+    if(CONFIG.debug) {
+      console.log("MSG", JSON.stringify(message.text));
+    }
+    let words = message.text.trim().split(" ");
+
+    function sendMessage(textMsg) {
+      if(CONFIG.debug) {
+        console.log("SENDING", textMsg, message.chat.id);
+      }
+
+      Shelly.call(
+        "HTTP.GET",
+        { 
+          url: "https://api.telegram.org/bot" + CONFIG.botKey + "/sendMessage?chat_id=" + message.chat.id + "&text=" + textMsg, 
```

Comment:
`"https://api.telegram.org/bot" + CONFIG.botKey + ` can be precomputed at start

Reply:
Done [1d5bad7](https://github.com/ALLTERCO/shelly-script-examples/pull/58/commits/1d5bad7a4031cfb0f255a819839310d8d9470b0d)